Source/src/modules/ModuleDebugMode.cpp

```diff
@@ -0,0 +1,156 @@
+
+UpdateStatus Hachiko::ModuleDebugMode::Update(const float delta)
+{
+	if (App->input->GetKey(SDL_SCANCODE_LALT) == Hachiko::KeyState::KEY_REPEAT &&
+		App->input->GetKey(SDL_SCANCODE_D) == Hachiko::KeyState::KEY_DOWN)
+	{
+		Toggle();
+	}
+
+	if (is_gui_active)
+	{
+		DrawGUI();
+	}
+
+	RenderGui();
+
+	return UpdateStatus::UPDATE_CONTINUE;
+}
+
+void Hachiko::ModuleDebugMode::RenderGui()
+{
+	ImGui::Render();
+	ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+
+	ImGui::UpdatePlatformWindows();
+	ImGui::RenderPlatformWindowsDefault();
+	SDL_GL_MakeCurrent(App->window->GetWindow(), App->renderer->GetGLContext());
+}
+
+UpdateStatus Hachiko::ModuleDebugMode::PostUpdate(const float delta)
+{
+	return UpdateStatus::UPDATE_CONTINUE;
+}
+
+bool Hachiko::ModuleDebugMode::CleanUp()
+{
+	return true;
+}
+
+ImGuiWindowFlags Hachiko::ModuleDebugMode::SetupWindow()
+{
+	ImGuiWindowFlags window_flags = ImGuiWindowFlags_NoDocking |
+		ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoTitleBar;
+	return window_flags;
+}
+
+void Hachiko::ModuleDebugMode::UpdateRenderValues()
+{
+	fps = EngineTimer::fps;
+	const Hachiko::RenderList* render_list = App->renderer->GetRenderList();
+	poly_on_screen = render_list->GetPolycountRendered();
+	poly_total = render_list->GetPolycountTotal();
+}
+
+const Hachiko::GameObject* Hachiko::ModuleDebugMode::FindPlayer()
+{
+	for (Hachiko::GameObject* child : App->scene_manager->GetRoot()->children)
+	{
+		if (child->name == "Player")
+		{
+			return child;
+		}
+	}
+}
+
+void Hachiko::ModuleDebugMode::DrawGUI()
+{
+	static SDL_version version;
+	static const float vram_total = hw_info.vram_capacity / 1024.0f;
+	float vram_free = hw_info.vram_free / 1024.0f;
+	float vram_usage = vram_total - vram_free;
+	float frame_rate = 0.0f;
+	float3 player_pos_editor = player->GetTransform()->GetPosition();
+	
+	window_flags = SetupWindow();
+	UpdateRenderValues();
+
+	if (ImGui::Begin("InGame Window", &is_gui_active, window_flags))
+	{
+		App->renderer->FpsGraph();
+		ImGui::Separator();
+
+		ImGui::Text("Polycount (On screen): %i", poly_on_screen);
+		ImGui::Text("Polycount (Total loaded): %i", poly_total);
+		ImGui::Separator();
+		ImGui::Text("CPUs: %d", hw_info.cpu);
+		ImGui::Text("System RAM: %.1f Gb", hw_info.ram);
+		ImGui::Text("GPU: %s", hw_info.gpu);
+		ImGui::Text("Vendor: %s", hw_info.gpu_vendor);
+		ImGui::Text("VRAM: %.1f Mb", vram_free);
+		ImGui::Text("Vram Usage:  %.1f Mb", vram_usage);
+		ImGui::Text("Vram Available:  %.1f Mb", vram_free);
+		ImGui::Separator();
+
+		if(ImGui::Button("Toggle Solid"))
+		{
+			glPolygonMode(GL_FRONT_AND_BACK, GL_FILL);
+		}
+		ImGui::SameLine();
+		if(ImGui::Button("Toggle Wire"))
+		{
+			glPolygonMode(GL_FRONT_AND_BACK, GL_LINE);
+		}
+		ImGui::Separator();
+
+		ImGui::Text("Player position: ");
+		if (ImGui::DragFloat3("(x, y, z)", player_pos_editor.ptr(), 0.1f, -inf, inf))
+		{
+			player->GetTransform()->SetGlobalPosition(player_pos_editor);
+		}
+	}
```

Comment:
We could add in a next PR buttons to enable-disable debug draw and some other options from the config window, but 100% optional right now.